packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -231,6 +234,27 @@ class DatadogRumPlugin(
                         result.missingParameter(call.method)
                     }
                 }
+                "updatePerformanceMetrics" -> {
+                    val buildTimes = call.argument<List<Double>>(PARAM_BUILD_TIMES)
+                    val rasterTimes = call.argument<List<Double>>(PARAM_RASTER_TIMES)
+                    if(buildTimes != null && rasterTimes != null) {
```

Comment:
why do we check both in the same condition and not independently?

Reply:
Simplicity more than anything. For the most part, these two parameters not being null is enforced by the type system in Dart, and contract checks in unit tests. If either one is null, it's a contract violation and reported to telemetry.  I could report which individual parameter violated the contract, but I haven't done that anywhere yet so this pattern is through the whole file.